hyperspy/models/model1d.py

```diff
@@ -710,15 +718,36 @@ def _model2plot(self, axes_manager, out_of_range2nans=True):
             ns[np.where(self.channel_switches)] = s
             s = ns
         return s
+    
+    def _residual_for_plot(self, axes_manager):
+        """From an model1D object, the original signal is subtracted 
+        by the model signal then returns the residual
+        """
+
+        old_axes_manager = None
+        if axes_manager is not self.axes_manager:
+            old_axes_manager = self.axes_manager
+            self.axes_manager = axes_manager
+            self.fetch_stored_values()
+
+        #Residual = Signal - model
+        s = (self.signal.__call__() - self.__call__(ignore_channel_switches = True))
+        if old_axes_manager is not None:
+            self.axes_manager = old_axes_manager
+            self.fetch_stored_values()
```

Comment:
This can be simplify to:

```suggestion
    def _residual_for_plot(self):
        """From an model1D object, the original signal is subtracted
        by the model signal then returns the residual
        """

        return self.signal.__call__() - self.__call__(ignore_channel_switches=True)

```
Github doesn't allow me to make a suggestion over a deleted line and `return s` below obviously also need to be removed in the above suggestion.

Reply:
I added a keyword argument to _residual_for_plot to prevent the error message attached below.
def _residual_for_plot(self,**kwargs):
![image](https://github.com/hyperspy/hyperspy/assets/112028355/8ff58b08-823b-4aba-897e-adfe5755532f)